src/util/currency.rs

```diff
@@ -10,3 +10,90 @@ pub fn parse_sum(sum: &str) -> Result<u64> {
 pub fn sum_to_string(sum: u64) -> String {
     sum.to_string()
 }
+
+pub enum Currency {
+    Sat,
+}
+
+impl Currency {
+    // Convert a string to a Currency enum
+    pub fn from_str(s: &str) -> Result<Self> {
+        match s.to_lowercase().as_str() {
+            "sat" => Ok(Currency::Sat),
+            _ => Err(Error::Validation(format!("Unsupported currency: {s}"))),
+        }
+    }
+
+    // Convert a Currency enum to a string
+    pub fn as_str(&self, currency: &str) -> Option<&'static str> {
+        match Self::from_str(currency) {
+            Ok(Currency::Sat) => Some("sat"),
+            _ => None,
+        }
+    }
+}
+
+// From satoshis to BTC
+pub fn sat_to_btc(sats: u64) -> f64 {
```

Comment:
Using f64 for currency conversion is problematic, because floating point ops can introduce precision errors.

To to this correctly, we'd need something like rust-decimal (https://docs.rs/rust_decimal/latest/rust_decimal/) or something similar.

Since we currently send the `sum` as a string, it would also be OK to convert to a BTC string like "0.100".